coderd/agentapi/stats_test.go

```diff
@@ -164,6 +171,15 @@ func TestUpdateStates(t *testing.T) {
 		// User gets fetched to hit the UpdateAgentMetricsFn.
 		dbM.EXPECT().GetUserByID(gomock.Any(), user.ID).Return(user, nil)
 
+		// Ensure that pubsub notifications are sent.
+		publishStats := make(chan bool)
+		ps.Subscribe(codersdk.WorkspaceNotifyChannel(workspace.ID), func(_ context.Context, description []byte) {
+			go func() {
+				publishStats <- bytes.Equal(description, []byte{})
+				close(publishStats)
+			}()
+		})
```

Comment:
This can panic if multiple messages are received. I think it would be better to just pipe the messages into a `chan []byte` here without closing, then do the comparison below. Also, by using a goroutine here you're not necessarily guaranteed order.